packages/graph/hash_graph/lib/graph/src/store/postgres/query/mod.rs

```diff
@@ -57,6 +60,13 @@ pub trait Path {
     ///
     /// [`terminating_table_name()`]: Self::terminating_table_name
     fn column_access(&self) -> ColumnAccess;
+
+    /// Returns the field if the path is provided by a user.
+    ///
+    /// One example of a user provided path is [`DataTypeQueryPath::Custom("custom string")`]
```

Comment:
What does this path actually mean on a data type? :thinking:

Reply:
The `Custom` variant is used to read `additionalProperties` from data types.

Reply:
Gotcha. I'm not entirely sure how I feel about this, because we _**do not**_ want `additionalProperties` on data types in the medium-term. (When we come up with non-primitive data types we want them fully specified like property types).

Is there a way we can better capture the purpose of this, so _we know to remove it when it becomes redundant_?

Reply:
We agreed offline, that we are going to remove the `Custom` field (in a separated PR).

Reply:
[Asana task](https://app.asana.com/0/0/1203225514907841/f) (_internal_)